kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -560,6 +574,67 @@ ColumnVector visitCoalesce(ScalarExpression coalesce) {
             );
         }
 
+        @Override
+        ColumnVector visitLike(final Like like) {
+            PredicateChildrenEvalResult argResults = evalBinaryExpressionChildren(like);
+            ColumnVector left = argResults.leftResult;
+            ColumnVector right = argResults.rightResult;
+            int numRows = argResults.rowCount;
+            boolean[] result = new boolean[numRows];
+            boolean[] nullability = new boolean[numRows];
+            for (int rowId = 0; rowId < numRows; rowId++) {
+                String leftOp = left.getString(rowId);
+                String rightOp = right.getString(rowId);
+                if (!Objects.isNull(leftOp) && !Objects.isNull(rightOp)) {
+                    nullability[rowId] = false;
+                    String rightOpRegexPattern =
+                            escapeLikeRegex(rightOp, like.getEscape());
+                    result[rowId] = leftOp.matches(rightOpRegexPattern);
+                } else {
+                    nullability[rowId] = true;
+                    result[rowId] = false;
+                }
+            }
+            return new DefaultBooleanVector(numRows,
+                    Optional.of(nullability), result);
```

Comment:
instead of evaluating the result for all rows at once, we can return a `ColumnVector` that lazily evaluates for the row needed. You can also look at `ElementAtEvaluator#eval`.

```
        return new ColumnVector() {

            @Override
            public DataType getDataType() {
                return BooleanType.BOOLEAN;
            }

            @Override
            public int getSize() {
                return left.getSize();
            }

            @Override
            public void close() {
                Utils.closeCloseables(left, right);
            }

            @Override
            public boolean isNullAt(int rowId) {
                return left.isNullAt(rowId) || right.isNullAt(rowId);
            }

            @Override
            public boolean getBoolean(int rowId) {
                // like expression result.
            }
        };
```

Reply:
done.